dotnet/csharp.bzl

```diff
@@ -474,20 +475,51 @@ csharp_autoconf = repository_rule(
     local = True,
 )
 
-def csharp_configure():
-  """Finds the mono and mcs binaries installed on the local system and sets
-  up an external repository to use the local toolchain.
-
-  To use the local Mono toolchain installed on your system, add the following
-  to your WORKSPACE file:
-
-  ```python
-  csharp_configure()
-  ```
-  """
-  csharp_autoconf(name = "local_config_csharp")
+def _mono_osx_repository_impl(repository_ctx):
+  # TODO(jwall): The below is necessary due to bug:
+  #   https://github.com/bazelbuild/bazel/issues/1235.
+  #   when that bug is fixed the below should be removed.
+  working_dir = repository_ctx.path("")
+  repository_ctx.file(repository_ctx.path("empty"))
+  if not working_dir.exists:
+    fail("working_dir %s does not exist! See https://github.com/bazelbuild/bazel/issues/1235" % working_dir)
+
+  download_output = repository_ctx.path("")
+  #download the package
+  repository_ctx.download_and_extract(
+    repository_ctx.attr._osx_pkg,
+    download_output,
+    repository_ctx.attr._osx_sha256,
+    "", "mono")
+
+  # now we create the build file.
+  toolchain_build = """
+package(default_visibility = ["//visibility:public"])
+exports_files(["mono", "mcs"])
```

Comment:
nit: on Linux there is sandboxing and you might need to declare a dependency on all the files of the framework. Not needed yet so you might ignore it.